platform/linux-generic/include/odp_pktio_ops_subsystem.h

```diff
@@ -85,12 +86,39 @@ typedef ODP_MODULE_CLASS(pktio_ops) {
 	odp_api_proto(pktio_ops, print) print;
 } pktio_ops_module_t;
 
-/* Maximum size of pktio specific ops data.*/
-#define ODP_PKTIO_ODPS_DATA_MAX_SIZE 80000
+/* Pktio ops data pool */
+#define ODP_PKTIO_OPS_DATA_POOL_SIZE 160000
+#define ODP_PKTIO_OPS_DATA_POOL_NAME "ODP_PKTIO_OPS_DATA"
 
 /* Extract pktio ops data from pktio entry structure */
 #define odp_ops_data(_p, _mod) \
-	((pktio_ops_ ## _mod ## _data_t *)(uintptr_t)_p->s.ops_data)
+	((pktio_ops_ ## _mod ## _data_t *)_p->s.ops_data)
+
+#define odp_ops_data_alloc(_p, _mod)					\
+({ \
+	odpdrv_shm_pool_t _pool = ODPDRV_SHM_POOL_INVALID;		\
+	uint64_t _size = sizeof(pktio_ops_ ## _mod ## _data_t);		\
+									\
+	_p->s.ops_data = NULL;			\
+	_pool = odpdrv_shm_pool_lookup(ODP_PKTIO_OPS_DATA_POOL_NAME);	\
+	if (_pool != ODPDRV_SHM_POOL_INVALID)				\
+		_p->s.ops_data = odpdrv_shm_pool_alloc(_pool, _size);	\
+									\
+	((pktio_ops_ ## _mod ## _data_t *)_p->s.ops_data);		\
```

Comment:
It will require a cast when is called. 
A pktio_type may implement another way to allocate memory starting form the name of the pool / is not mandatory to use those macros but are helpful for existing pktio types.

Reply:
@joseppc   Btw, 'odp_ops_data_alloc(_p, _mod)' vs. odp_ops_data_alloc(_p, _size) ?

Reply:
ah, yes, true, I didn't think about this detail...